java/vector/src/main/java/org/apache/arrow/vector/BitVector.java

```diff
@@ -344,6 +344,21 @@ public final void get(int index, NullableBitHolder holder) {
       holder.isSet = 1;
       holder.value = get(index);
     }
+
+    /**
+     * Get the number of bits set to 1
+     * @return the number of bits set to 1
+     */
+    public final int getNullCount() {
+      int count = 0;
+      for (int i = 0; i < allocationSizeInBytes; ++i) {
+        byte byteValue = data.getByte(i);
+        // Java uses two's complement binary representation, hence 11111111_b which is -1 when converted to Int
+        // will have 32bits set to 1. Masking the MSB and then adding it back solves the issue.
+        count += Integer.bitCount(byteValue & 0x7F) - (byteValue >> 7);
+      }
+      return count;
```

Comment:
Let's do this at long level. We can add two methods, this one which is validcount and another which is count - this which is null count.